spiketoolkit/postprocessing/postprocessing_tools.py

```diff
@@ -620,24 +768,108 @@ def export_to_phy(recording, sorting, output_folder, nPC=3, electrode_dimensions
+            writer = csv.writer(tsvfile, delimiter='\t', lineterminator='\n')
+            writer.writerow(['cluster_id', 'chan_grp'])
+            for i, u in enumerate(sorting.get_unit_ids()):
+                writer.writerow([i, sorting.get_unit_property(u, 'group')])
     else:
-        max_num_chans_in_group = recording.get_num_channels()
-        channel_groups = np.array([0] * recording.get_num_channels())
-    if nPC > max_num_chans_in_group:
-        nPC = max_num_chans_in_group
-        if verbose:
-            print("Changed number of PC to number of channels: ", nPC)
+        with (output_folder / 'cluster_channel_group.tsv').open('w') as tsvfile:
+            writer = csv.writer(tsvfile, delimiter='\t', lineterminator='\n')
+            writer.writerow(['cluster_id', 'ch_group'])
+            for i, u in enumerate(sorting.get_unit_ids()):
+                writer.writerow([i, 0])
+
+    np.save(str(output_folder / 'amplitudes.npy'), amplitudes)
+    np.save(str(output_folder / 'spike_times.npy'), spike_times.astype('int64'))
+    np.save(str(output_folder / 'spike_templates.npy'), spike_templates.astype('int64'))
+    np.save(str(output_folder / 'spike_clusters.npy'), spike_clusters.astype('int64'))
+    np.save(str(output_folder / 'pc_features.npy'), pc_features)
+    np.save(str(output_folder / 'pc_feature_ind.npy'), pc_feature_ind.astype('int64'))
+    np.save(str(output_folder / 'templates.npy'), templates)
+    np.save(str(output_folder / 'template_ind.npy'), templates_ind.astype('int64'))
+    np.save(str(output_folder / 'similar_templates.npy'), similar_templates)
+    np.save(str(output_folder / 'channel_map.npy'), channel_map.astype('int64'))
+    np.save(str(output_folder / 'channel_map_si.npy'), channel_map_si.astype('int64'))
+    np.save(str(output_folder / 'channel_positions.npy'), positions)
+    np.save(str(output_folder / 'channel_groups.npy'), channel_groups)
+
+    if save_waveforms:
+        np.save(str(output_folder / 'waveforms.npy'), np.array(waveforms))
+
+    if verbose:
+        print('Saved phy format to: ', output_folder)
+        print('Run:\n\nphy template-gui ', str(output_folder / 'params.py'))
+
+
+def _compute_templates_similarity(templates):
+    similarity = np.zeros((len(templates), len(templates)))
+    for i, t_i in enumerate(templates):
+        for j, t_j in enumerate(templates):
+            t_i_lin = t_i.reshape(t_i.shape[0] * t_i.shape[1])
+            t_j_lin = t_j.reshape(t_j.shape[0] * t_j.shape[1])
+            a = np.corrcoef(t_i_lin, t_j_lin)
+            similarity[i, j] = np.abs(a[0, 1])
+    return similarity
+
+
+def _compute_whitening_and_inverse(recording):
+    white_recording = st.preprocessing.whiten(recording)
+    wh_mat = white_recording._whitening_matrix
+    wh_mat_inv = np.linalg.inv(wh_mat)
+    return wh_mat, wh_mat_inv
+
+
+def _get_random_spike_waveforms(recording, sorting, unit, max_num, snippet_len, channels=None):
```

Comment:
I think we could use the get_unit_waveforms for this. We can just return, optionally, the indexes. The get_waveforms also takes care of splitting by group, etc.., so it's more powerful.

Reply:
I think this is used inside get_unit_waveforms... I didn't write it

Reply:
I see! then it's been just moved :) looks good